lib/DeclarePushConstantsPass.cpp

```diff
@@ -62,6 +62,13 @@ bool DeclarePushConstantsPass::runOnModule(Module &M) {
     PushConstants.emplace_back(clspv::PushConstant::GlobalOffset);
   }
 
+  if ((clspv::Option::Language() ==
+       clspv::Option::SourceLanguage::OpenCL_C_20) ||
+      (clspv::Option::Language() ==
+       clspv::Option::SourceLanguage::OpenCL_CPP)) {
```

Comment:
This should also check if get_enqueued_local_size is used. Seems unnecessary to unconditionally require this value.

Reply:
Yes, agree. This is a problem for `get_work_dim()` as well and I thought I'd tackle that as a separate PR but I don't mind doing it here if that's the preference.

Reply:
I'm ok with a separate PR.

Reply:
Ok, I'll make that my next PR then.